### ISSUE TITLE
Improving Training Efficiency

### ISSUE THREAD

**Issue by Kohze** [Kohze/TensorFlowBoard-Word2Vec#1]
It seems that the training efficiency is rather slow - after 12 epocs we are still at ~5% accuracy.

**Comment by vaskonov** [comment]
The `word2vec.py` script is the original tensorflow script from the [official repository](https://github.com/tensorflow/models) with the following changes:
1) The deprecated `mul` method was replaced with `multiply`.
2) `save_vocab` was changed to meet the [tensorflow embedding visualization requirement](https://www.tensorflow.org/get_started/embedding_viz).
3) `projector` has been added to encapsulate words into embedding.
The training and the evaluation methods remained untouched.

According to the [Issue 1092](https://github.com/tensorflow/models/issues/1092) at the tensorflow repository, the word2vec.py publicly available version is different from the tensorflow word2vec.py internal version with recent updates.
I am following up the issue with the tensorflow team.
@Kohze, by the way, at the late epocs of training there was a meaningful improvement in accuracy?

**Comment by Kohze** [comment]
Yes the comparison to the example shows that we didnt do anything that should decrease the training efficiency. Now I tested a 40 epic run - and while the training loss function decrease as expected the accuracy is randomly flunctuating around 5% +- 0.5 - therefore no improvement in late epocs.
